### DISCUSSION TITLE
NIST - FIPS 203, 204 for CRYSTALS-Kyber and CRYSTALS-Dilithium

### DISCUSSION THREAD

**Discussion by wagner-robert** [open-quantum-safe/oqs-provider#487] [General]
All, glancing through the algorithms it appears the code treats CRYSTALS-Kyber and ML-KEM as two different algorithms. The same can be said for CRYSTALS-Dilithium and ML-DSA. NIST approved FIPS 203 and 204 on Aug-13-2024 and just calls them ML-KEM and ML-DSA now.  I was wondering if there are plans to collapse these into single algorithms?

**Comment by baentsch** [comment]
Yes. As far as oqsprovider users are concerned, this will happen "invisibly" (maybe with the exception of Kyber/Dilithium disappearing from the list of supported algs). See https://github.com/orgs/open-quantum-safe/discussions/1892.
